sklearn/utils/estimator_checks.py

```diff
@@ -787,19 +787,48 @@ def check_estimators_unfitted(name, Estimator):
     with warnings.catch_warnings(record=True):
         est = Estimator()
 
-    assert_raises(NotFittedError, est.predict, X)
-
     if hasattr(est, 'predict'):
-        assert_raises(NotFittedError, est.predict, X)
+        check_unfitted(name, est.predict, X)
 
     if hasattr(est, 'decision_function'):
-        assert_raises(NotFittedError, est.decision_function, X)
+        check_unfitted(name, est.decision_function, X)
 
     if hasattr(est, 'predict_proba'):
-        assert_raises(NotFittedError, est.predict_proba, X)
+        check_unfitted(name, est.predict_proba, X)
 
     if hasattr(est, 'predict_log_proba'):
-        assert_raises(NotFittedError, est.predict_log_proba, X)
+        check_unfitted(name, est.predict_log_proba, X)
+
+
+def check_unfitted(name, func, *args, **kwargs):
+    """Helper function to check that an AttributeError or a ValueError
+    is raised when calling 'func' on unfitted estimator
+        Parameters
+        ----------
+        name : string
+            Name of the estimator
+
+        func : callable
+            Calable object to raise error if estimator is unfitted
+
+        *args : the positional arguments to `func`.
+
+        **kw : the keyword arguments to `func`
+    """
+    correct_raise = True
+    try:
+        func(*args, **kwargs)
+        correct_raise = False
+    except (ValueError, AttributeError):  # correct Error
+        pass
+    except:
+        correct_raise = False
```

Comment:
This could be simplified as follows:

```python
correct_raise = False
try:
    func(*args, **kwargs)
except (ValueError, AttributeError):  # correct Error
    correct_raise = True

if not correct_raise:
    ...
```